pkg/operators/v1alpha1/types_test.go

```diff
@@ -208,3 +210,72 @@ func TestCatalogSource_Poll(t *testing.T) {
 		require.Equal(t, tt.result, table[i].catsrc.Poll(), table[i].description)
 	}
 }
+
+func TestUpdateStrategyUnmarshal(t *testing.T) {
+	type TestStruct struct {
+		UpdateStrategy UpdateStrategy `json:"updateStrategy,omitempty"`
+	}
+	validDuration, err := time.ParseDuration("45m")
+	if err != nil {
+		panic(fmt.Errorf("error parsing duration: %s", err))
+	}
+	defaultDuration, err := time.ParseDuration("15m")
+	if err != nil {
+		panic(fmt.Errorf("error parsing duration: %s", err))
+	}
+	tests := []struct {
+		name string
+		in   []byte
+		out  TestStruct
+		err  error
+	}{
+		{
+			name: "valid",
+			in:   []byte(`{"UpdateStrategy": {"registryPoll":{"interval":"45m"}}}`),
+			out: TestStruct{
+				UpdateStrategy{
+					&RegistryPoll{
+						RawInterval:  "45m",
+						Interval:     &metav1.Duration{Duration: validDuration},
+						ParsingError: "",
+					},
+				},
+			},
+		},
+		{
+			name: "invalid",
+			in:   []byte(`{"UpdateStrategy": {"registryPoll":{"interval":"19mError Code"}}}`),
+			out: TestStruct{
+				UpdateStrategy{
+					&RegistryPoll{
+						RawInterval:  "19mError Code",
+						Interval:     &metav1.Duration{Duration: defaultDuration},
+						ParsingError: "error parsing spec.updateStrategy.registryPoll.interval. Using the default value of 15m0s instead. Error: time: unknown unit \"mError Code\" in duration \"19mError Code\"",
+					},
+				},
+			},
+		},
+		{
+			name: "empty",
```

Comment:
nit: should we add a test case if an empty JSON blob is sent?

Reply:
Is an empty JSON blob for an `Unmarshaller` func ever an input? The unmarshaller func only gets called when there is valid data to unmarshall (i.e if you don't specify `UpdateStrategy` in your CatalogSource, this function is not getting invoked by the complier anyway)

Reply:
I see, thank you for the clarification.